tests/integration/sandbox.py

```diff
@@ -130,9 +133,10 @@ def init(self):
         LOG.info('Cluster created: %s', dbid)
         return self
 
-    def join(self, addresses):
+    def join(self, addresses, expect_error=None):
```

Comment:
I think there is not enough separation of concerns here between utility functions and test cases. An alternative can be to simply expect the exception when calling `join`, and maybe pass an explicit `retries=1` to save time.

Reply:
hmm, ok, that could make sense.  retries=1 and try/except the error in the caller.  I can do that.

Reply:
think I've done this now